common/scala/src/whisk/common/ConsulKVReporter.scala

```diff
@@ -26,27 +32,24 @@ import spray.json.JsString
  */
 class ConsulKVReporter(
     kvStore: ConsulKV,
-    initialDelayMilli: Int,
-    periodMilli: Int,
+    initialDelay: FiniteDuration,
+    interval: FiniteDuration,
     hostKey: String,
     startKey: String,
     statusKey: String,
     updater: () => Map[String, JsValue]) {
 
-    private val t = new Thread() {
-        override def run() = {
-            Thread.sleep(initialDelayMilli)
-            val (selfHostname, stderr, exitCode) = SimpleExec.syncRunCmd(Array("hostname", "-f"))(TransactionId.unknown)
-            kvStore.put(hostKey, JsString(selfHostname))
-            kvStore.put(startKey, JsString(s"${DateUtil.getTimeString}"))
-            while (true) {
-                kvStore.put(statusKey, JsString(s"${DateUtil.getTimeString}"))
-                updater().foreach({ case (k, v) => kvStore.put(k, v) })
-                Thread.sleep(periodMilli)
-            }
+    implicit val system = ActorSystem()
```

Comment:
Shouldn't we take the `system` as constructor argument? It's not recommended to have multiple actor systems so it should share the system with other components to the extent applicable (in tests it doesn't matter).